hummingbot/connector/exchange/kucoin/kucoin_exchange.py

```diff
@@ -351,13 +363,75 @@ async def _update_trading_fees(self):
             trading_pair = await self.trading_pair_associated_to_exchange_symbol(symbol=fee_json["symbol"])
             self._trading_fees[trading_pair] = fee_json
 
+    async def _update_orders_fills(self, orders: List[InFlightOrder]):
+        # This method in the base ExchangePyBase, makes an API call for each order.
+        # Given the rate limit of the API method and the breadth of info provided by the method
+        # the mitigation proposal is to collect all orders in one shot, then parse them
+        # Note that this is limited to 50000 orders
+        # An alternative for Kucoin would be to use the limit/fills that returns 24hr updates, which should
+        # be sufficient, the rate limit seems better suited
+        all_trades_updates: List[TradeUpdate] = []
+        try:
+            all_trades_updates: List[TradeUpdate] = await self._all_trades_updates(orders)
+        except asyncio.CancelledError:
+            raise
+        except Exception as request_error:
+            self.logger().warning(
+                f"Failed to fetch trade updates. Error: {request_error}")
+
+        for trade_update in all_trades_updates:
+            self._order_tracker.process_trade_update(trade_update)
+
+    async def _all_trades_updates(self, orders: List[InFlightOrder]) -> List[TradeUpdate]:
+        trade_updates: List[TradeUpdate] = []
+        exchange_to_client = {o.exchange_order_id: {"client_id": o.client_order_id, "trading_pair": o.trading_pair} for o in orders}
+
+        # We don't need updates for trades created earlier than last_update
+        last_update_ms: int = int(1000 * max(self._last_order_fills_request_ts_s, min([o.creation_timestamp for o in orders])))
+        self._last_order_fills_request_ts_s = time.time()
+
+        all_fills_response = await self._api_get(
+            path_url=CONSTANTS.FILLS_PATH_URL,
+            params={
+                "pageSize": 500,
+                "endAt": last_update_ms,
```

Comment:
do you check if start_at is the older ts? i.e. start time 100,  it will give you the trades that the ts is greater than 100, or how it works? because that changes the logic a lot

Reply:
Kucoin:"If you only specified the start time, the system will automatically calculate the end time (end time = start time + 7 * 24 hours)"
So what are we trying to achieve? Do we want all the fills that happened since the last time we checked? the last or the first trade that was created in the list we pass? Would we not have duplicate info if we overlap the calls?

Reply:
The new PR should address all these improvements